app/src/main/java/com/kms/cura/view/CustomSpinner.java

```diff
@@ -0,0 +1,67 @@
+package com.kms.cura.view;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.widget.Spinner;
+
+/**
+ * Created by toanbnguyen on 7/18/2016.
+ */
+public class CustomSpinner extends Spinner {
+
+    private OnSpinnerEventsListener mListener;
+    private boolean mOpenInitiated = false;
+    private boolean mToggleFlag = true;
+
+    public CustomSpinner(Context context) {
+        super(context);
+    }
+
+    public CustomSpinner(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    @Override
+    public boolean performClick() {
+        mOpenInitiated = true;
+        if (mListener != null) {
+            mListener.onSpinnerOpened();
+        }
+        mToggleFlag = false;
+        boolean result = super.performClick();
+        mToggleFlag = true;
```

Comment:
Why set `mToggleFlag = false` above then set it back to `mToggleFlag = true;` below ?